propertime/time.py

```diff
@@ -0,0 +1,860 @@
+
+    def __sub__(self, other):
+        raise NotImplementedError('Cannot subtract anything from a TimeUnit. Only a TimeUnit from something else.')
+
+    def __truediv__(self, other):
+        raise NotImplementedError('Division for TimeUnits is not implemented')
+
+    def __rtruediv__(self, other):
+        raise NotImplementedError('Division for TimeUnits is not implemented')
+
+    def __mul__(self, other):
+        raise NotImplementedError('Multiplication for TimeUnits is not implemented')
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __eq__(self, other):
+
+        # Check against another TimeUnit
+        if isinstance(other, TimeUnit):
+            if self.is_calendar() and other.is_calendar():
+                # Check using the calendar components
+                if self.years != other.years:
+                    return False
+                if self.months != other.months:
+                    return False
+                if self.weeks != other.weeks:
+                    return False
+                if self.days != other.days:
+                    return False
+                if self.hours != other.hours:
+                    return False
+                if self.minutes != other.minutes:
+                    return False
+                if self.seconds != other.seconds:
+                    return False
+                if self.microseconds != other.microseconds:
+                    return False
+                return True
+            elif self.is_calendar() and not other.is_calendar():
+                return False
+            elif not self.is_calendar() and other.is_calendar():
+                return False
+            else:
+                # Check using the duration in seconds, as 15m and 900s are actually the same unit
+                if self.as_seconds() == other.as_seconds():
+                    return True
+
+        # Check for direct equality with value, i.e. comparing with a string
+        if self.value == other:
+            return True
+
+        # Check for equality on the same "registered" value
+        if isinstance(other, TimeUnit):
+            if self.value == other.value:
+                return True
+
+        # Check for duration as seconds equality, i.e. comparing with a float
+        try:
+            if self.as_seconds() == other:
+                return True
+        except (TypeError, ValueError):
+            # Raised if this or the other other TimeUnit is of calendar type, e.g.
+            # ValueError: You can ask to get a calendar TimeUnit as seconds only if you provide the unit starting point
+            pass
+
+        # If everything fails, return false:
+        return False
+
+    def _is_composite(self):
+        types = 0
+        for item in self._mapping_table:
+            if getattr(self, self._mapping_table[item]): types +=1
+        return True if types > 1 else False 
+
+    @property
+    def type(self):
```

Comment:
timeunits are mutable, so the "type" can change at runtime. Is this useful or just risky?

I'd use different constructors (at least) which provide different objects.

Also, the user doesn't control duration type initialization. Maybe I want a physical duration, but I make a mistake and I get a calendar item. I'd prefer a constructor like "Duration.physical()..." and if I pass non-physical things there, I get an error.